### ISSUE TITLE
bug: parser unrecoverably fails if there is a `#+TODO: (...)` directive

### ISSUE THREAD

**Issue by whacked** [xiaoxinghu/orgajs#5]
```javascript
const { parse } = require("orga")
parse("#+TODO: (whatever)\n* something");
```
will throw
```
TypeError: Cannot read property 'value' of undefined
    at _parse (./orgajs/node_modules/orga/lib/inline.js:101:19)
    at parse (./orgajs/node_modules/orga/lib/inline.js:25:10)
    at Parser.process (./orgajs/node_modules/orga/lib/processors/headline.js:49:32)
    at Parser.parseSection (./orgajs/node_modules/orga/lib/parser.js:96:24)
    at Parser.process (./orgajs/node_modules/orga/lib/processors/blank.js:9:15)
    at Parser.parseSection (./orgajs/node_modules/orga/lib/parser.js:96:24)
    at Parser.process (./orgajs/node_modules/orga/lib/processors/keyword.js:38:15)
    at Parser.parseSection (./orgajs/node_modules/orga/lib/parser.js:96:24)
    at Parser.parse (./orgajs/node_modules/orga/lib/parser.js:80:15)
    at parse (./orgajs/node_modules/orga/lib/index.js:13:17)
```
This is unrecoverable, meaning that after getting this error, the parser will continue to throw the same error on any string with a headline, such as `parse("* something")`. This issue persists even if the `orga` is re-required after removing the require cache using `delete require.cache[require.resolve("orga")]`

I have not chased this very far and don't understand why the error persists across reimporting, but the `undefined` comes from somewhere around
https://github.com/xiaoxinghu/orgajs/blob/master/packages/orga/src/lexer.js#L116
where `pattern` for matching `headline` gets set to a regex containing `(whatever)` and basically fails for all headlines.

Thanks for your work.

**Comment by xiaoxinghu** [comment]
it is caused by the `()` in the `TODO` directive. Normally we don't expect to see non-alphanumeric characters in todo keywords, but it's a risk to break the whole parser. It's fixed in v0.5.3, please verify. Thanks for the good finding.

**Comment by whacked** [comment]
Thanks for the quick update. I can verify that this issue is fixed on windows (node v9.5.0) and on linux (v10.6.0). On linux (v8.0.0) the parser fails with

```
./orgajs/node_modules/orga/lib/inline.js:63
        throw _iteratorError;
        ^

SyntaxError: Invalid regular expression: /(.*?(?:[\s\({'"]|^))\*([^,'"\s].*?)(?<=[^,'"\s])\*((?:[\s-\.,:!?'\)}]|$).*)/: Invalid group
    at RegExp (<anonymous>)
    at markup (./orgajs/node_modules/orga/lib/inline.js:21:10)
    at _loop (./orgajs/node_modules/orga/lib/inline.js:36:19)
    at parse (./orgajs/node_modules/orga/lib/inline.js:51:7)
    at Parser.process (./orgajs/node_modules/orga/lib/processors/headline.js:49:32)
    at Parser.parseSection (./orgajs/node_modules/orga/lib/parser.js:96:24)
    at Parser.parse (./orgajs/node_modules/orga/lib/parser.js:80:15)
    at parse (./orgajs/node_modules/orga/lib/index.js:13:17)
    at Object.<anonymous> (./orgajs/orga-test.js:3:13)
    at Module._compile (module.js:569:30)
```

for orga >= 0.5.2; though, this is likely a separate issue

**Comment by xiaoxinghu** [comment]
@whacked will try that out when I got chance. Thanks a lot.
